### PR TITLE
ChakraCore CI support

### PR DESCRIPTION
**Pull request by joaocgreis**
##### Checklist
- [x] `make -j4 test` (UNIX), or `vcbuild test nosign` (Windows) passes
- [x] commit message follows commit guidelines
##### Affected core subsystem(s)

Test
##### Description of change

This PR adds support to specify tests as flaky/skip/... independently by JavaScript engine.

The first two commits add the logic to detect the `jsEngine` in `test.py` when running the tests, and use that variable when reading the status files. The `--engine` argument of `test.py` is no longer necessary, so this reverts parts of https://github.com/nodejs/node-chakracore/commit/42bc2848a89a9f9ee145f9fdfe057465c410b2d9 and https://github.com/nodejs/node-chakracore/commit/50398dab52ad8c03745f0869cd632ab0f8c3d09e where it was added. The `jsEngine` variable is now passed to the test suites in a way that follows the change introduced in https://github.com/nodejs/node-chakracore/commit/edaf7af30bddef699ea68726fedd7a28d1dc999a .

The last commit adds an initial list of tests that are failing with ChakraCore on CI, gathered from several CI runs.

@nodejs/node-chakracore the CI job to test node-chakracore is now ready to be used: https://ci.nodejs.org/view/All/job/chakracore-test-windows/

CI for this PR: https://ci.nodejs.org/view/All/job/chakracore-test-windows/20/

CI fails because Visual C++ Build Tools are not yet supported, this is being tracked in https://github.com/nodejs/node-chakracore/issues/24 . All VS2015 builds are green.

### CODE REVIEW

File: test/parallel/parallel.status [hunk 1]
```diff
@@ -36,3 +36,34 @@ test-debug-signal-cluster                : PASS, FLAKY
 
 #covered by https://github.com/nodejs/node/issues/8271
 test-child-process-fork-dgram            : PASS, FLAKY
+
+[$jsEngine==chakracore]
```
Comment (kunalspathak) [inline test/parallel/parallel.status:40]: Can we differentiate this based on architecture? For x86 we have 21 failures while for x64 we have 25 failures. Below list seems to be taken for x64?
x86 failures in parallel folder

``` cmd
parallel/test-crypto-dh
parallel/test-fs-stat
parallel/test-http-same-map
parallel/test-promises-warning-on-unhandled-rejection
parallel/test-repl
parallel/test-repl-mode
parallel/test-promises-unhandled-rejections
parallel/test-repl-tab-complete
parallel/test-string-decoder
parallel/test-util-inspect-simd
parallel/test-util-inspect-proxy
parallel/test-vm-context
parallel/test-vm-create-and-run-in-context
parallel/test-vm-cached-data
parallel/test-vm-global-identity
parallel/test-vm-low-stack-space
parallel/test-vm-preserves-property
parallel/test-vm-proxies
parallel/test-vm-strict-mode
parallel/test-vm-timeout
```

x64 failures in addition to x86 failures:

``` cmd
parallel/test-buffer-includes
parallel/test-buffer-indexof
parallel/test-buffer-slow
```

Comment (joaocgreis) [reply test/parallel/parallel.status:40]: @kunalspathak updated. I moved the `buffer` tests to a x64-only subgroup. The list of tests on your comment above is missing a few tests that I included in this PR (e.g. `parallel/test-force-repl-with-eval`, that I took from the failure that happened here: https://ci.nodejs.org/view/All/job/node-test-binary-windows/RUN_SUBSET=3,VS_VERSION=vs2015,label=win2008r2/4171/console).

**Comment by digitalinfinity** [comment]
Thanks @joaocgreis - this is awesome. Looks good to me- maybe @orangemocha or someone with node test experience should also sign off though?

File: tools/test.py [hunk 2]
```diff
@@ -1564,12 +1560,19 @@ def Main():
           print "Can't determine the arch of: '%s'" % vm
           print archEngineContext.stderr.rstrip()
           continue
+        jsEngineContext = Execute([vm, "-p", "process.jsEngine"], context)
+        jsEngine = jsEngineContext.stdout.rstrip()
+        if jsEngineContext.exit_code is not 0 or jsEngine == "undefined":
```
Comment (thefourtheye) [inline tools/test.py]: Don't use `is` operator to check the values. Use `!=`

**Comment by joaocgreis** [comment]
Updated. Thanks @thefourtheye !

**Comment by kunalspathak** [comment]
LGTM

File: tools/test.py [hunk 3]
```diff
@@ -1560,16 +1556,23 @@ def Main():
...
+        jsEngine = jsEngineContext.stdout.rstrip()
+        if jsEngineContext.exit_code != 0 or jsEngine == "undefined":
+          print "Can't determine the jsEngine of: '%s'" % vm
+          print jsEngineContext.stderr.rstrip()
+          continue
         env = {
           'mode': mode,
           'system': utils.GuessOS(),
           'arch': vmArch,
+          'jsengine': jsEngine, # variable names are lowercased
```
Comment (thefourtheye) [inline tools/test.py]: Can you please explain what the comment means? Are you referring to `jsengine`?

Comment (joaocgreis) [reply tools/test.py]: @thefourtheye yes, that explains the case difference. But the comment was far from clear, updated with a better one. PTAL

**Comment by joaocgreis** [comment]
CI: https://ci.nodejs.org/view/All/job/chakracore-test-windows/20/

Landed in https://github.com/nodejs/node-chakracore/commit/c7c08e5c75ed57169ec6e5c17011ba5c2ab816d5 , https://github.com/nodejs/node-chakracore/commit/297350bb85de707bf51267e09134702857927784 and https://github.com/nodejs/node-chakracore/commit/c8817a2bf09b198be509eac590de093daf2f6ab3 .

**Pull request closed without merging**